src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -0,0 +1,163 @@
+    bearingToPoint
+} from '../math/flightMath';
+import { nm } from '../utilities/unitConverters';
+
+/**
+ * Loop through `waypointModelList` and determine where along the route an
+ * aircraft should spawn
+ *
+ * @function _calculateSpawnPositions
+ * @param waypointModelList {array<StandardWaypointModel>}
+ * @param spawnOffsets {array}
+ * @return spawnPositions {array}
+ */
+export const _calculateSpawnPositions = (waypointModelList, spawnOffsets, airport) => {
+    const spawnPositions = [];
+
+    // for each new aircraft
+    for (let i = 0; i < spawnOffsets.length; i++) {
+        let spawnOffset = spawnOffsets[i];
+
+        // for each fix ahead
+        for (let j = 1; j < waypointModelList.length; j++) {
+            const waypoint = waypointModelList[j];
+
+            if (waypoint.distanceFromPreviousWaypoint > spawnOffset) {
+                // if point before next fix
+                const nextFix = waypoint;
+                const previousFix = waypointModelList[j - 1];
+                const heading = bearingToPoint(previousFix.gpsXY, nextFix.gpsXY);
+                const spawnPoint = fixRadialDist(previousFix.gps, heading, spawnOffset);
+                const position = PositionModel.calculatePosition(spawnPoint, airport.position, airport.magnetic_north);
+                // TODO: this looks like it should be a model object
+                const preSpawnHeadingAndPosition = {
+                    heading,
+                    position,
+                    nextFix: nextFix.name
+                };
+
+                spawnPositions.push(preSpawnHeadingAndPosition);
+
+                break;
+            }
+
+            // if point beyond next fix subtract distance from spawnOffset and continue
+            spawnOffset -= waypoint.distanceFromPreviousWaypoint;
+        }
+    }
+
+    return spawnPositions;
+};
+
+/**
+ * Calculate distrance(s) from center where an aircraft should exist onload or airport change
+ *
+ * @function _assembleSpawnOffsets
+ * @param entrailDistance {number}
+ * @param totalDistance {number}
+ * @return spawnOffsets {array<number>}
+ */
+export const _assembleSpawnOffsets = (entrailDistance, totalDistance) => {
+    const spawnOffsets = [];
+
+    // distance between successive arrivals in nm
+    for (let i = entrailDistance; i < totalDistance; i += entrailDistance) {
+        spawnOffsets.push(i);
+    }
+
+    return spawnOffsets;
+};
+
+/**
+ * Calculate heading, nextFix and position data to be used when creating an
+ * `AircraftInstanceModel` along a route.
+ *
+ * @function _preSpawn
+ * @param spawnPatternJson
+ * @param navigationLibrary
+ * @param airport
+ * @return {array<object>}
+ */
+export const _preSpawn = (spawnPatternJson, navigationLibrary, airport) => {
+    // find last fix along STAR that is outside of airspace, ie: next fix is within airspace
+    // distance between closest fix outside airspace and airspace border in nm
+    let extra = 0;
```

Comment:
reference the above comment, why not just rename `extra` to `distanceFromClosestFixToAirspaceBoundary`? It's long, but at least descriptive.